src/rest-server/src/routes/v2/job.js

```diff
@@ -18,31 +18,21 @@
 
 // module dependencies
 const express = require('express');
-const token = require('../middlewares/token');
-const jobController = require('../controllers/job');
-const jobParam = require('../middlewares/jobV2');
+const token = require('../../middlewares/token');
+const protocol = require('../../middlewares/v2/protocol');
+const asyncHandler = require('../../middlewares/v2/asyncHandler');
+const controller = require('../../controllers/v2/job');
+
 
 const router = new express.Router();
 
 router.route('/')
-    /** GET /api/v2/jobs - Get list of jobs */
-    .get(jobParam.query, jobController.list)
-
-    /** POST /api/v2/jobs - Update job */
-    .post(token.check, jobParam.submission, jobController.init, jobController.update);
-
-router.route('/:jobName')
-    /** GET /api/v2/jobs/:jobName - Get job status */
-    .get(jobController.get)
-
-    /** PUT /api/v2/jobs/:jobName - Update job */
-    .put(token.check, jobParam.submission, jobController.update)
-
-    /** DELETE /api/v2/jobs/:jobName - Remove job */
-    .delete(token.check, jobController.remove);
-
-/** Load job when API with jobName route parameter is hit */
-router.param('jobName', jobController.load);
+  /** POST /api/v2/jobs - Update job */
+  .post(
+    asyncHandler(protocol.wrap),
```

Comment:
I think apply `asyncHandler` in the controller file is better, since the actual exports function in the controller file is not a  express handler we want.

Reply:
updated